alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetWorkflowAlertsAction.kt

```diff
@@ -221,12 +224,20 @@ class TransportGetWorkflowAlertsAction @Inject constructor(
             val queryBuilder = QueryBuilders.boolQuery()
             queryBuilder.must(QueryBuilders.termsQuery("_id", associatedAlertIds))
             queryBuilder.must(QueryBuilders.termQuery(Alert.STATE_FIELD, Alert.State.AUDIT))
-            val searchRequest = SearchRequest(alertIndex)
-            searchRequest.source().query(queryBuilder)
-            val response: SearchResponse = client.suspendUntil { search(searchRequest, it) }
+            val scroll = Scroll(TimeValue.timeValueMinutes(1L))
+            val scrollSize = 100 // Number of hits to return per scroll
+            var searchRequest = SearchRequest(alertIndex)
+            searchRequest.source().query(queryBuilder).size(scrollSize)
+            searchRequest.scroll(scroll)
+            var response: SearchResponse = client.suspendUntil { search(searchRequest, it) }
             associatedAlerts.addAll(parseAlertsFromSearchResponse(response))
+            val scrollId = response.scrollId
+            while (response.hits.hits.isNotEmpty()) {
+                response = client.suspendUntil { client.searchScroll(SearchScrollRequest(scrollId), it) }
+                associatedAlerts.addAll(parseAlertsFromSearchResponse(response))
+            }
```

Comment:
Do we want to limit this and have page number responses in the api response so the customer can paginate? This would be dangerous if there were many associated alerts like from doc level monitors.

Reply:
we can paginate only on the primary field - alerts
we cannot paginate on the thee associated alerts field as it provides information associated to primary resource.
Open to ideas on how to handle this
Pagination requires statefulness which we provide through search query out of the box. Associated alerts is a list we compile from the alerts parsed from the primary search query response. we can’t paginate on that

Reply:
I think if the user provides the alert id, then they can paginate through the associated alerts. If they do not provide the alert id, it would paginate through the main alerts only then.